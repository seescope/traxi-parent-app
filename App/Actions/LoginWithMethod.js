import { NativeModules } from 'react-native';

export default () => dispatch =>
  NativeModules.Authentication.authenticate()
  .then(profile => dispatch({ type: 'LOGGED_IN', profile }))
