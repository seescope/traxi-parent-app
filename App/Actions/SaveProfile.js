import { AsyncStorage } from 'react-native';
import Firebase from 'firebase';

const saveProfileToFirebase = newProfile => new Promise((resolve, reject) => {
  const firebase = new Firebase(
    `https://traxiapp.firebaseio.com/parents/${newProfile.UUID}`,
  );
  firebase.set(newProfile, error => {
    if (error) {
      reject(error);
    } else {
      resolve();
    }
  });
});

const saveProfileToStorage = profile =>
  AsyncStorage.setItem('profile', JSON.stringify(profile));

const saveProfile = profile => 
    Promise.all([
      saveProfileToStorage(profile),
      saveProfileToFirebase(profile),
    ]);

export default saveProfile;
