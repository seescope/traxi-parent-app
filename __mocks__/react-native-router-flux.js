import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
