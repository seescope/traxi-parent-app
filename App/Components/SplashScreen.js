/* eslint global-require: "off" */

import React from 'react';
import { Dimensions, View, Text, Image } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import * as Animatable from 'react-native-animatable';

import Background from '../Components/Background';
import Button from '../Components/Button';
import Spacing from '../Components/Spacing';
import { WHITE, TRANSPARENT } from '../Constants/Colours';
import loginWithMethod from '../Actions/LoginWithMethod';
import { logError } from '../Utils';

const { height } = Dimensions.get('window');
const handleError = error => {
  logError(`Error logging in: ${error}`);
  alert('There was an error logging you in. Please try again.');
};

const logoStyle = {
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: 28,
  textAlign: 'center',
  fontFamily: 'Raleway-ExtraBold',
  marginBottom: 16,
};

const containerStyle = {
  flex: 1,
  alignItems: 'center',
  paddingHorizontal: 32,
};

const imageStyle = {
  height: height / 2,
};

const SplashScreen = () =>
  <Background style={containerStyle}>
    <Animatable.Image
      useNativeDriver
      resizeMode="contain"
      style={imageStyle}
      animation="bounceInDown"
      source={require('../Images/trail_preview.png')}
    />

    <Spacing height={32} />

    <Animatable.Text
      useNativeDriver
      animation="bounceIn"
      delay={1000}
      style={logoStyle}
    >
      Start seeing what your kids are doing.
    </Animatable.Text>

    <Spacing height={32} />

    <Animatable.View
      useNativeDriver
      animation="bounceInUp"
      delay={2000}
    >
      <Button onPress={() => Actions.intro()}>Show me how</Button>
    </Animatable.View>
  </Background>;


SplashScreen.propTypes = {
  getStarted: React.PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  getStarted: () => dispatch(loginWithMethod())
    .then(() => Actions.setName())
    .catch(handleError),
});

export default connect(null, mapDispatchToProps)(SplashScreen);
