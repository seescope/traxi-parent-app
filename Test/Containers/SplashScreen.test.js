jest.mock(
  '../../App/Actions/LoginWithMethod',
  () => jest.fn(() => Promise.resolve()),
);
jest.mock(
  '../../App/Actions/SaveProfile',
  () => jest.fn(() => Promise.resolve()),
);

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import LoginWithMethod from '../../App/Actions/LoginWithMethod';
import SaveProfile from '../../App/Actions/SaveProfile';
import SplashScreen, {
  mapDispatchToProps,
} from '../../App/Containers/SplashScreen';

const mockStore = configureStore();
const testStore = mockStore(() => {});

it('renders the <SplashScreen> component', () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <SplashScreen />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('handles logins', () => {
  const mockDispatch = jest.fn(() => Promise.resolve());
  return mapDispatchToProps(mockDispatch).getStarted().then(() => {
    expect(mockDispatch.mock.calls).toMatchSnapshot();
    expect(Actions.intro).toHaveBeenCalled();
    expect(LoginWithMethod).toHaveBeenCalled();
    expect(SaveProfile).toHaveBeenCalled();
  });
});
