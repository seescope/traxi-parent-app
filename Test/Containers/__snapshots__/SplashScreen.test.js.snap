// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles logins 1`] = `
Array [
  Array [
    Promise {
      "_45": 0,
      "_54": null,
      "_65": undefined,
      "_81": 1,
    },
  ],
]
`;

exports[`renders the <SplashScreen> component 1`] = `
<View
  colors={
    Array [
      "#448AFF",
      "#90CAF9",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingHorizontal": 32,
    }
  }
>
  <Image
    animation="bounceInDown"
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 667,
      }
    }
    useNativeDriver={true}
  />
  <View
    style={
      Object {
        "height": 32,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    animation="bounceIn"
    delay={1000}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 28,
        "marginBottom": 16,
        "textAlign": "center",
      }
    }
    useNativeDriver={true}
  >
    What are your children doing online?
  </Text>
  <View
    style={
      Object {
        "height": 32,
      }
    }
  />
  <View
    animation="bounceInUp"
    delay={2000}
    useNativeDriver={true}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }
      >
        Find out now
         
      </Text>
    </View>
  </View>
</View>
`;
